catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/GazetteerQueryOfflineSolr.java

```diff
@@ -0,0 +1,465 @@
+
+    this.client = clientFactory.newClient(GazetteerConstants.STANDALONE_GAZETTEER_CORE_NAME);
+    startupBuilderExecutor.submit(this::pingAndInitializeSuggester);
+  }
+
+  @Override
+  public List<GeoEntry> query(String queryString, int maxResults) throws GeoEntryQueryException {
+    SolrQuery solrQuery =
+        new SolrQuery(String.format("title_txt:\"%s\"", ClientUtils.escapeQueryChars(queryString)));
+    solrQuery.setRows(Math.min(maxResults, GazetteerQueryOfflineSolr.MAX_RESULTS));
+
+    QueryResponse response = null;
+    try {
+      response = client.query(solrQuery, METHOD.POST);
+    } catch (SolrServerException | IOException e) {
+      throw new GeoEntryQueryException("Error while querying", e);
+    }
+    return response
+        .getResults()
+        .stream()
+        .map(this::transformMetacardToGeoEntry)
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public GeoEntry queryById(String id) throws GeoEntryQueryException {
+    SolrQuery solrQuery =
+        new SolrQuery(String.format("id_txt:\"%s\"", ClientUtils.escapeQueryChars(id)));
+    solrQuery.setRows(1);
+
+    QueryResponse response = null;
+    try {
+      response = client.query(solrQuery, METHOD.POST);
+    } catch (SolrServerException | IOException e) {
+      throw new GeoEntryQueryException("Error while querying by ID", e);
+    }
+
+    return response
+        .getResults()
+        .stream()
+        .map(this::transformMetacardToGeoEntry)
+        .findFirst()
+        .orElseThrow(() -> new GeoEntryQueryException("Could not find id"));
+  }
+
+  @Override
+  public List<Suggestion> getSuggestedNames(String queryString, int maxResults)
+      throws GeoEntryQueryException {
+    SolrQuery solrQuery = new SolrQuery();
+    solrQuery.setRequestHandler("/suggest");
+    solrQuery.setParam(SUGGEST_Q, ClientUtils.escapeQueryChars(queryString));
+    solrQuery.setParam(SUGGEST_DICT, SUGGEST_DICT_VALUE);
+    solrQuery.setParam("suggest.count", Integer.toString(Math.min(maxResults, MAX_RESULTS)));
+
+    QueryResponse response;
+    try {
+      response = client.query(solrQuery);
+    } catch (SolrServerException | IOException e) {
+      throw new GeoEntryQueryException("Error while querying", e);
+    }
+
+    return Optional.ofNullable(response)
+        .map(QueryResponse::getSuggesterResponse)
+        .map(SuggesterResponse::getSuggestions)
+        .map(suggestionsPerDict -> suggestionsPerDict.get(SUGGEST_DICT_VALUE))
+        .orElse(Collections.emptyList())
+        .stream()
+        .map(suggestion -> new SuggestionImpl(suggestion.getPayload(), suggestion.getTerm()))
+        .collect(Collectors.toList());
+  }
+
+  private void pingAndInitializeSuggester() {
+    Failsafe.with(
+            new RetryPolicy()
+                .retryWhen(false)
+                .withMaxDuration(5, TimeUnit.MINUTES)
+                .withBackoff(100, 3_000, TimeUnit.MILLISECONDS))
```

Comment:
❓ This is probably exhaustively fine which means when flakiness occurs, this failure will be a very rare sight. Would it be an improvement to use `whenAvailable` and an `Initializer` instead? 

https://github.com/codice/ddf/blob/master/platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java#L133

Drawback there is that, if it turns out Solr isn't available for awhile, we will never log a "timeout" message claiming our "initialization" (if you include async tasks) didn't "finish" (it technically did because we booted up) unless we can provide a timeout. So I'm not sure.

Reply:
So i just got rid of this code/complexity all together since we have a independant gazetteer suggester I'm building it on startup so this logic was not needed anymore 😄